theano/tensor/signal/pool.py

```diff
@@ -2369,13 +2378,13 @@ def c_code(self, node, name, inp, out, sub):
                   r_st[%(i)s] -= pd[%(i)s];
                   r_end[%(i)s] -= pd[%(i)s];
                   // handle the case where no padding, ignore border is True
-                  if (%(ignore_border)s)
+                  if (%(params)s->ignore_border)
                   {
                     r_end[%(i)s] = r_end[%(i)s] > r[%(i)s] ? r[%(i)s] : r_end[%(i)s];
                   }
                   // use the index to find the correct position in the output
                   o_idx[%(non_pool_ndim)s + %(i)s] = r_idx[%(i)s];
-            """ % dict(i=i, ignore_border=ignore_border, non_pool_ndim=non_pool_ndim)
```

Comment:
Missing `params=sub['params']` for this dict, as `%(params)s->ignore_border` is used in the formatted string.